packages/gatekeeper/src/components/login.tsx

```diff
@@ -1,34 +1,71 @@
 import React from 'react';
+import { OauthLoginProps } from '..';
+import { NO_PROVIDERS_STRING, OAUTH_LOGIN_PROMPT_STRING } from '../helpers/constants';
 import { getProviderFromOptions, Providers } from '../helpers/oauth';
 
-/** interface for OauthLogin props */
-export interface OauthLoginProps {
-  ProviderLinksComponent?: React.ElementType;
+/** types of supported authorization grant flow */
+export enum AuthorizationGrantType {
+  IMPLICIT = 'IMPLICIT',
+  AUTHORIZATION_CODE = 'AUTHORIZATION_CODE'
+}
+
+/** describes object to pass to the OAuthHook */
+export interface OAuthLoginHookOptions {
   providers: Providers;
+  authorizationGrant: AuthorizationGrantType;
 }
 
 /** interface for ProviderLinks props */
-export interface ProviderLinksProps {
-  providers: Providers;
+export interface ProviderLinksProps extends OAuthLoginHookOptions {
+  OAUTH_LOGIN_PROMPT?: string;
```

Comment:
I think we should maintain normal/regular case rules for object property names (even if it refers to something that is populated from a constant)